src/Codeception/Lib/Connector/Laravel.php

```diff
@@ -463,4 +463,36 @@ public function clearApplicationHandlers(): void
     {
         $this->applicationHandlers = [];
     }
+    
+    /**
+     * Make sure files are \Illuminate\Http\UploadedFile instances with the private $test property set to true.
+     * Fixes issue https://github.com/Codeception/Codeception/pull/3417.
+     *
+     * @param array $files
+     * @return array
+     */
+    protected function filterFiles(array $files)
+    {
+        $files = parent::filterFiles($files);
+        return $this->convertToTestFiles($files);
+    }
+
+    /**
+     * @param array $files
+     * @return array
+     */
```

Comment:
btw, this docblock does not add additional info regarding the method signature, so it can be removed.